stubs/networkx/networkx/convert_matrix.pyi

```diff
@@ -0,0 +1,70 @@
+from _typeshed import Incomplete
+from collections.abc import Callable, Iterable
+from typing import TypeVar, overload
+from typing_extensions import Literal, TypeAlias
+
+import numpy
+from networkx.classes.graph import Graph, _Node
+
+# stub_uploader won't allow pandas-stubs in the requires field https://github.com/typeshed-internal/stub_uploader/issues/90
+# from pandas import DataFrame
+_DataFrame: TypeAlias = Incomplete
+# from pandas.core.dtypes.base import ExtensionDtype
+_ExtensionDtype: TypeAlias = Incomplete
+_G = TypeVar("_G")
+
+def to_pandas_adjacency(
+    G: Graph[_Node],
+    nodelist: list[_Node] | None = None,
+    dtype: numpy.dtype[Incomplete] | None = None,
+    order: Literal["C", "F"] | None = None,
+    multigraph_weight: Callable[[Iterable[float]], float] = ...,
+    weight: str = "weight",
+    nonedge: float = 0.0,
+) -> _DataFrame: ...
+@overload
+def from_pandas_adjacency(df: _DataFrame, create_using: type[_G]) -> _G: ...
+@overload
+def from_pandas_adjacency(df: _DataFrame, create_using: None = None) -> Graph[Incomplete]: ...
+def to_pandas_edgelist(
+    G: Graph[_Node],
+    source: str | int = "source",
+    target: str | int = "target",
+    nodelist: list[_Node] | None = None,
+    dtype: _ExtensionDtype | None = None,
+    edge_key: str | int | None = None,
+) -> _DataFrame: ...
+@overload
+def from_pandas_edgelist(
+    df: _DataFrame,
+    source: str | int = "source",
+    target: str | int = "target",
+    edge_attr: str | int | Iterable[str | int] | Literal[True] | None = None,
+    *,
+    create_using: type[_G],
+    edge_key: str | None = None,
+) -> _G: ...
+@overload
+def from_pandas_edgelist(
+    df: _DataFrame,
+    source: str | int = "source",
+    target: str | int = "target",
+    edge_attr: str | int | Iterable[str | int] | Literal[True] | None = None,
+    create_using: None = None,
+    edge_key: str | None = None,
+) -> Graph[Incomplete]: ...
+def to_numpy_array(
+    G: Graph[_Node],
+    nodelist: list[_Node] | None = None,
+    dtype: numpy.dtype[Incomplete] | None = None,
+    order: Literal["C", "F"] | None = None,
+    multigraph_weight: Callable[[Iterable[float]], float] = ...,
+    weight: str = "weight",
+    nonedge: float = 0.0,
+) -> numpy.ndarray[Incomplete, numpy.dtype[Incomplete]]: ...
+@overload
+def from_numpy_array(
+    A: numpy.ndarray[Incomplete, Incomplete], parallel_edges: bool = False, create_using: None = ...
+) -> Graph[Incomplete]: ...
+@overload
+def from_numpy_array(A: numpy.ndarray[Incomplete, Incomplete], parallel_edges: bool = False, *, create_using: type[_G]) -> _G: ...
```

Comment:
`: None = None` should be fine
This won't support `from_numpy_array(np.ndarray([]), False, Graph)` because `create_using` is keyword only
```suggestion
    A: numpy.ndarray[Incomplete, Incomplete], parallel_edges: bool = False, create_using: None = None
) -> Graph[Incomplete]: ...
@overload
def from_numpy_array(A: numpy.ndarray[Incomplete, Incomplete], parallel_edges: bool = False, *, create_using: type[_G]) -> _G: ...
@overload
def from_numpy_array(A: numpy.ndarray[Incomplete, Incomplete], parallel_edges: bool, create_using: type[_G]) -> _G: ...
```
Yes it's annoying that you have to add 2 overloads just for that. See: https://mail.python.org/archives/list/python-dev@python.org/message/7W5RMJVBRN6NE6A3LUP5Y4BMOZKQRYWH/ and https://discuss.python.org/t/some-improvements-to-function-overloads/36275/2?u=avasam